### PR TITLE
Update safety check and cryptography version

### LINKED ISSUE #3682: Remove safety ignore rule for pyca/cryptography
## Description
Initially suggested by @heartsucker in https://github.com/freedomofpress/securedrop/pull/3679#issuecomment-409645593:
https://github.com/freedomofpress/securedrop/commit/6ba90dac7fd76cd97f58ef48f629b8931378689e sets a rule to ignore CVE-2018-10903. 

We should remove this ignore rule as soon as possible (e.g. as soon as we update the dependency in SecureDrop). This task and discussions surrounding the rationale behind ignoring this check is detailed/tracked in https://github.com/freedomofpress/securedrop/issues/3677

## User Stories

As a human, disabling and/or ignoring security warning/alerts/notifications is a bad idea.

### PR DESCRIPTION
**Pull request by heartsucker**
## Status

Ready for review

## Description of Changes

Fixes #3682

- Remove the rule ignoring the out of date version of `cryptography`
- Run the `make update-pip-requirements` script to bump cryptography and other dependencies

## Testing

CI should catch any incompatibilities in the libraries

## Deployment

No special considerations.

### If you made changes to the server application code:

- [x] Linting (`make ci-lint`) and tests (`make -C securedrop test`) pass in the development container

### CODE REVIEW

File: securedrop/dockerfiles/xenial/Dockerfile [hunk 1]
```diff
@@ -26,7 +26,8 @@ RUN curl -LO https://launchpad.net/~ubuntu-mozilla-security/+archive/ubuntu/ppa/
 RUN gem install sass -v 3.4.23
 
 COPY requirements requirements
-RUN pip install -r requirements/securedrop-app-code-requirements.txt && \
+RUN pip install -U setuptools==40.8.0 && \
```
[the remark this answers is not in the document]
Comment (emkll) [reply securedrop/dockerfiles/xenial/Dockerfile]: Per https://github.com/freedomofpress/securedrop/issues/3441#issuecomment-390060677 , the setuptools version bundled in Xenial should be sufficient to use the latest cryptography library and may clash with system-provided setuptools package.

[the remark this answers is not in the document]
Comment (heartsucker) [reply securedrop/dockerfiles/xenial/Dockerfile]: Ok, so we could remove it from the Xenial build. But no, we shoudn't put it in requirements. This is considered a bad practice. I'll wait for others to comment on the main review comment before pushing a change.

**Comment by emkll** [comment]
So to recap the discussions above and discussions during standup, as part of this PR, we should ensure that the nightly cron job `staging-with-rebase-trusty` is removed, since it will reliably fail. Since `release/0.12.1` branch uses the Trusty staging logic by default, so any PR to 0.12.1 will have CI against Trusty VMs (and that new installs should always be using Xenial at this point), I propose we forgo a nightly Trusty CI staging job.

Based on my previous comments, I think we should also, if possible, not install setuptools in Xenial dev image. I think it's fine to keep it in the Trusty environment to ensure compatibility with the latest `cryptography` package.

**Comment by heartsucker** [comment]
Someone can append/rebase those changes if they need it in by Monday.

**Comment by emkll** [comment]
This is now ready for re-review, I've removed the Trusty cron target and did minimal cleanup (scoped to circle config). To avoid scope creep in this PR I propose we open a follow-up to delete the now dead platform conditional CI logic (primarily bash scripts) used for Trusty/Xenial support.

**Pull request merged**